test/test_tensordict.py

```diff
@@ -733,40 +725,34 @@ def test_lock(self, td_name, device):
 
     def test_lock_write(self, td_name, device):
         td = getattr(self, td_name)(device)
-        # TODO Fix once _items method is implemented for SubTensorDict
-        if td_name in ["sub_td", "sub_td2"]:
-            pytest.skip(
-                "Cannot use TensorDictKeysView for SubTensorDict instances at the"
-                "moment, skipping test case!!"
-            )
         td.lock()
         td_clone = td.clone()
         assert not td_clone.is_locked
         td_clone = td.to_tensordict()
         assert not td_clone.is_locked
         assert td.is_locked
         td = td.select(inplace=True)
-        for key, item in td_clone.items(True):
+        keys_view = _TensorDictKeysView(td_clone, include_nested=True, leaves_only=False, error_on_loop=False)
+        for key in keys_view:
+            item = td_clone.get(key)
             with pytest.raises(RuntimeError, match="Cannot modify locked TensorDict"):
                 td.set(key, item)
         td.unlock()
-        for key, item in td_clone.items(True):
+        keys_view = _TensorDictKeysView(td_clone, include_nested=True, leaves_only=False, error_on_loop=False)
```

Comment:
I think you can reuse the `keys_view` you created above

Reply:
Correct!!